src/Toml/Type/Value.hs

```diff
@@ -116,7 +116,9 @@ instance (t ~ 'TText) => IsString (Value t) where
 instance Eq (Value t) where
     (Bool b1)    == (Bool b2)    = b1 == b2
     (Integer i1) == (Integer i2) = i1 == i2
-    (Double f1)  == (Double f2)  = f1 == f2
+    (Double f1)  == (Double f2)
+        | isNaN f1 && isNaN f2 = True
```

Comment:
Okay, I see the problem why it doesn't work. The problem is in the fact that in `TOML` data type we actually store `AnyValue`. And `Eq` instance for `AnyValue` is written like this:

https://github.com/kowainik/tomland/blob/543ae70c960134da49852c757b4b96e810490cf5/src/Toml/Type/AnyValue.hs#L34

So actual check is not triggered. The problem is in the fact that `Eq` for `AnyValue` doesn't reuse `Eq` for `Value`. This should be fixed. Unfortunately, implementation like:

```haskell
AnyValue val1 == AnyValue val2 = val1 == val2
```

doesn't work because `val1` and `val2` may have different types.

You need to reuse `sameValue` function in the `Eq` instance for `AnyValue`:

https://github.com/kowainik/tomland/blob/543ae70c960134da49852c757b4b96e810490cf5/src/Toml/Type/Value.hs#L200

Reply:
Now tests has passed! Thank you!